merino/providers/weather/provider.py

```diff
@@ -52,14 +74,71 @@ async def initialize(self) -> None:
     def hidden(self) -> bool:  # noqa: D102
         return False
 
+    def cache_key_for_weather_report(self, geolocation: Location) -> Optional[str]:
+        """Compute a Redis key used to look up and store cached weather reports for a location."""
+        cache_inputs = self.backend.cache_inputs_for_weather_report(geolocation)
+        if not cache_inputs:
+            return None
+
+        return f"{self.name}:v1:report:{hashlib.blake2s(cache_inputs).hexdigest()}"
+
+    async def fetch_cached_weather_report(
+        self, geolocation: Location
+    ) -> Optional[WeatherReport]:
+        """Fetch a cached weather report, if available, for a location.
+
+        Raises:
+            CacheMissError: if there's no entry in the cache for this location.
+            CacheEntryError: if the cached weather report can't be deserialized.
+        """
+        cache_key: Optional[str] = self.cache_key_for_weather_report(geolocation)
+        if not cache_key:
+            raise CacheMissError
+
+        cache_value: Optional[bytes] = await self.redis.get(cache_key)
+        if not cache_value:
+            raise CacheMissError
+
+        try:
+            weather_report_dict = json.loads(cache_value)
+            if not weather_report_dict:
+                return None
+            return WeatherReport.parse_obj(weather_report_dict)
+        except ValueError as exc:
```

Comment:
`Model.parse_*` will throw `ValidationError` if the validation fails. Let's catch that as well. 
```suggestion
        # from pydantic import ValidationError
        except (ValidationError, ValueError) as exc:
```

Reply:
`ValueError` is a superclass of both `pydantic.ValidationError` and `json.JSONDecodeError`! 😄

Reply:
Ah, I see. Could you add a quick comment to note that for future reference?

Reply:
Done!